modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlApiBaseTest.java

```diff
@@ -360,18 +360,28 @@ private void checkMixedTransactions(Matcher<String> planMatcher) {
     }
 
     private void checkTx(Session ses, boolean readOnly, boolean commit, boolean explicit, Matcher<String> planMatcher) {
-        Transaction outerTx = explicit ? (readOnly ? igniteTx().begin(new TransactionOptions().readOnly(true)) : igniteTx().begin()) : null;
+        Transaction outerTx = explicit ? igniteTx().begin(new TransactionOptions().readOnly(readOnly)) : null;
 
-        String query = "SELECT VAL0 FROM TEST ORDER BY VAL0";
+        String queryRo = "SELECT VAL0 FROM TEST ORDER BY VAL0";
 
-        assertQuery(outerTx, query).matches(planMatcher).check();
+        assertQuery(outerTx, queryRo).matches(planMatcher).check();
 
-        ResultSet<SqlRow> rs = executeForRead(ses, outerTx, query);
+        ResultSet<SqlRow> rs = executeForRead(ses, outerTx, queryRo);
 
         assertEquals(ROW_COUNT, asStream(rs).count());
 
         rs.close();
 
+        String queryRw = "UPDATE TEST SET VAL0=VAL0+1";
+        if (explicit && readOnly) {
+            assertThrowsSqlException(Sql.STMT_VALIDATION_ERR, "DML can't be run within read only transaction.",
```

Comment:
Do we have a test, which checks if failed DML rollback explicit transaction?
Maybe we can have such check right here, WDYT?

Reply:
We don't rollback transaction in case validation error